### PR TITLE
Audio rewrite

### PR DESCRIPTION
**Pull request by GrandPanda**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature thoroughly
  * I mean it probably works

**Issues Fixed:** I didn't like internal audio stuff

### Changes Proposed in this Pull Request
* Restructured internal audio 
* Moved user voice state into VoiceState object
* Added audio receive (with public API implemented by Austin in audio-receive)
* Moved roles and nicks to User map to match voice states

### CODE REVIEW

File: src/main/java/sx/blah/discord/api/internal/DiscordVoiceWS.java [hunk 1]
```diff
@@ -210,24 +145,15 @@ private void send(GatewayPayload payload) {
 		}
 	}
 
-	private void send(String message) {
+	public void send(String message) {
 		if (getSession() != null && getSession().isOpen()) {
 			getSession().getRemote().sendStringByFuture(message);
 		} else {
-			Discord4J.LOGGER.warn(LogMarkers.VOICE_WEBSOCKET, "Attempt to send message on closed session: {}", message);
+			Discord4J.LOGGER.warn(LogMarkers.WEBSOCKET, "Attempt to send message on closed session: {}", message);
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/api/internal/DiscordVoiceWS.java]: Should this not have the VOICE_WEBSOCKET log marker?

File: src/main/java/sx/blah/discord/handle/audio/impl/AudioManager.java [hunk 2]
```diff
@@ -71,74 +62,80 @@ public IAudioProcessor getAudioProcessor() {
...
+
+	@Override
+	public synchronized void unsubscribeReceiver(IAudioReceiver receiver) {
+		// Check general receivers
+		generalReceivers.removeIf(r -> r.equals(receiver));
+		// Check user receivers
+		userReceivers.values().forEach(list -> list.removeIf(r -> r.equals(receiver)));
+	}
+
+	public byte[] sendAudio() { //TODO: Audio padding
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/handle/audio/impl/AudioManager.java]: You should probably synchronize this

File: src/main/java/sx/blah/discord/handle/audio/impl/AudioManager.java [hunk 3]
```diff
@@ -71,74 +62,80 @@ public IAudioProcessor getAudioProcessor() {
...
 		IAudioProcessor processor = getAudioProcessor();
 		IAudioProvider provider = useProcessor ? processor : getAudioProvider();
 
 		return getAudioDataForProvider(provider);
 	}
 
+	public void receiveAudio(byte[] opusAudio, IUser user) {
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/handle/audio/impl/AudioManager.java]: And this

File: src/main/java/sx/blah/discord/handle/impl/obj/Embed.java [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import sx.blah.discord.api.internal.json.objects.EmbedObject;
 import sx.blah.discord.handle.obj.IEmbed;
 
-import java.awt.Color;
+import java.awt.*;
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/handle/impl/obj/Embed.java]: Nice import optimization :eyes:

File: src/main/java/sx/blah/discord/api/internal/Services.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import java.net.URL;
+import java.nio.file.Files;
+import java.util.*;
+import java.util.jar.JarOutputStream;
+import java.util.zip.ZipEntry;
+
+public class Services {
+	static void load() {
+		try {
+			File servicesJar = Files.createTempFile("discord4j-services", "jar").toFile();
```
Comment (quanticc) [inline src/main/java/sx/blah/discord/api/internal/Services.java]: "jar" should be ".jar" To create a file ending with .jar

File: src/main/java/sx/blah/discord/handle/impl/obj/VoiceChannel.java [hunk 6]
```diff
@@ -341,7 +337,7 @@ public IVoiceChannel copy() {
 
 	@Override
 	public List<IUser> getConnectedUsers() {
-		return parent.getUsers().stream().filter((user) -> user.getConnectedVoiceChannels().contains(this)).collect(Collectors.toList());
+		return guild.getUsers().stream().filter(u -> u.getVoiceStateForGuild(guild).getChannel() != null && u.getVoiceStateForGuild(guild).getChannel().equals(this)).collect(Collectors.toList());
```
Comment (quanticc) [inline src/main/java/sx/blah/discord/handle/impl/obj/VoiceChannel.java]: Maybe swap it for:

```java
this.equals(u.getVoiceStateForGuild(guild).getChannel())
```

**Pull request merged**